### PR TITLE
[TGUI] Stack Craft 2: Restart

### PR DESCRIPTION
**Pull request by AyIong**
## What Does This PR Do
Maybe TM Required, for a few rounds
Revives that PR - https://github.com/ParadiseSS13/Paradise/pull/23253
Most of the work was done by @Gaxeer 
I just made an asset and finished the UI as he originally wanted it.

The changes made after the closed PR are minimal, among them:
1. Added result preview for all recipes, made on aset so that it is generated only 1 time, at the first opening.
2. Partial revert - https://github.com/ParadiseSS13/Paradise/pull/24252
Since the original PR was made without counting on this strange and in my opinion illogical move, I thought it is better to roll it back
3. Added an additional parameter to the Collapsible component, which is responsible for the content style, it does not affect the already existing Collapsible

## Why It's Good For The Game
Convenient material crafting, with search and preview, and not returning to the top after any action

## Images of changes
The first opening is quite long, as there is a pulling of the results of the crafting, and their icons

https://github.com/ParadiseSS13/Paradise/assets/69762909/4849942e-9263-474e-b753-fe1fc325f2fe

## Testing
Opened up the craft menu on a bunch of materials, crafted a bunch of shit.

<details>
<summary>Looked up the generated asset in the cache</summary>

asset 937bf69714f1ea3b7cdd84a7df0233bf

</details>

## Changelog
:cl: Gaxeer, Aylong
add: TGUI for stack crafting (metal sheets etc.)
/:cl:

### CODE REVIEW

File: code/modules/asset_cache/assets/asset_stack_recipes.dm
Comment (AyIong) [inline code/modules/asset_cache/assets/asset_stack_recipes.dm:1]: imo, cursed file

**Comment by FunnyMan3595** [comment]
Based on the video, that UI takes a *remarkably* long time to open. I'm not comfortable with anything over 0.5s, and that's around 4 times that.

**Comment by AyIong** [comment]
> Based on the video, that UI takes a _remarkably_ long time to open. I'm not comfortable with anything over 0.5s, and that's around 4 times that.

Only first time is a long open. Cause assets generating

**Comment by FunnyMan3595** [comment]
> > Based on the video, that UI takes a _remarkably_ long time to open. I'm not comfortable with anything over 0.5s, and that's around 4 times that.
> 
> Only first time is a long open. Cause assets generating

It should be able to open without the assets complete, and load them later, then

**Comment by AyIong** [comment]
> > > Based on the video, that UI takes a _remarkably_ long time to open. I'm not comfortable with anything over 0.5s, and that's around 4 times that.
> > 
> > 
> > Only first time is a long open. Cause assets generating
> 
> It should be able to open without the assets complete, and load them later, then

Impossible. Assets are made so that generation/receiving comes first, and then the rest of it.
I could remove the assets, but then not only the first opening of the UI would be quite long. I can also remove preview completely, but then the whole (or at least most of) the point is lost.
This is, as far as I know, the heaviest asset on para at this moment

**Comment by AyIong** [comment]
If someone knew how to decreace generation time, i'll be appreciated

**Comment by AyIong** [comment]
I hope someobe in future can optimize it

**Pull request closed without merging**